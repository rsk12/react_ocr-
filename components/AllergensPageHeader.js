import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Header, Icon } from 'react-native-elements';

const styles = StyleSheet.create({
  header: {},
  navBar: {},
  title: {},
  buttons: {},
  buttonPrimary: {},
});

const AddAllergenButton = (props) => (
  <Icon
    name='add-circle'
    onPress={props.onPress}
  />
);

export default class AllergensPageHeader extends React.Component {
  _handleAdd() {
    console.log("_handleAdd invoked!")
  }

  render() {
    return (
      <Header
        leftComponent={<View/>}
        centerComponent={{ text: 'Allergens', style: { color: '#fff' } }}
        rightComponent={<AddAllergenButton onPress={this._handleAdd}/>}
      />
    );
  }
}
