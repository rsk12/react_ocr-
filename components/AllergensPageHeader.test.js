import React from 'react';
import AllergensPageHeader from './AllergensPageHeader';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create( <AllergensPageHeader/> ).toJSON();
  expect(rendered).toBeTruthy();
});

it('renders correctly', () => {
      // given jest snapshot data
      const snapshotData = [{
        name: 'Foo'
      }];

      // when component is rendered
      const rendered = renderer.create( <AllergensPageHeader data = {
          snapshotData
        }
        /> ).toJSON();

        // then rendered tree matches snapshot
        expect(rendered).toMatchSnapshot();
      });
