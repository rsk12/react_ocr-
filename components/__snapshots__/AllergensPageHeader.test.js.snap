// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#476DC5",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 1,
        "height": 70,
        "padding": 15,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  >
    <View
      onPress={undefined}
      style={
        Array [
          false,
          false,
          false,
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      underlayColor="white"
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 24,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Allergens
    </Text>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 24,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
